File: day5/array.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        
        int count[26] ={0}; //initialised to 0 // as characters are less thus array used else unordered map would have been used
        
        for(char c: s)      // iterating the character array
            count[c-'a'] +=1; // if value is a then ascii value becomes 0 as 'a-a' // here count array is ready for checking uniqueness of character occurrence
        
        
        //for traversing and need to return the index not character
        for(int i =0; i<s.length(); ++i)
            if(count[s[i]- 'a']==1)
                return i;
        
        return -1;
        
    }
};

File: day5/array.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        count=[0]*26
        
        for c in s:     #iterating the character array
            count[ord(c)- ord('a')] += 1 
            # if value is a then ascii value becomes 0 as 'a-a' // here count array is ready for checking uniqueness of character occurrence # moreover directly ascii values cant be subtracted so first ascii value of given charater is found and then subbtracted a from it
        
        
        #for traversing and need to return the index not character
        for i in range(len(s)):
            if(count[ord(s[i])- ord('a')]==1):
                return i
        
        return -1;